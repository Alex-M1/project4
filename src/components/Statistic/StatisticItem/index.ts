import { connect } from 'react-redux';
import { STATISTIC_DATA } from '../../../../__mocks__/statisticByUUID';
import StatisticItem from './StatisticItem';

const mapStateToProps = () => ({
  statistic: STATISTIC_DATA,
});

export default connect(mapStateToProps)(StatisticItem);
