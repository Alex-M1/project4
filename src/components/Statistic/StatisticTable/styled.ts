import styled from 'styled-components';
import { IStyled } from 'common/types/styledTypes';
import { changeTheme } from 'src/helpers/changeTheme';

export const StWrapper = styled.div <IStyled>`
  width: 40%;
  height: 75%;
  border-radius: 12px;
  overflow: hidden;
  color: ${(props) => changeTheme(props, 'textColor')};
  background: ${(props) => changeTheme(props, 'backgroundSecondary')};
  box-shadow: 2px 2px 5px ${(props) => changeTheme(props, 'gray')};
`;
export const StHeaderWrapper = styled.div <IStyled>`
  width: 100%;
  height: 40px;
  margin: 10px 0;
  
  & h1 {
    padding: 5px 0;
    text-align: center;
  }

  & div {
    display: flex;
    justify-content: space-around;
    width: 85%;

    height: 30px;
    margin: 10px;

    & span:first-child {
      text-align: center;
    }
  }
`;
export const StResultsWrapper = styled.div <IStyled>`
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  max-height: 455px;
  margin-top: 40px;
  overflow: scroll;

  & div {
    display: flex;
    justify-content: flex-start;
    align-items: center;
    width: 90%;
    min-height: 20px;
    height: auto;
    margin: 5px 0;
    padding: 15px 0;
    border-bottom: 1px solid ${(props) => changeTheme(props, 'gray')};
    & span {
      text-align: center;
    }

    & span:first-child {
      width: 45%;
    }
  }
`;
