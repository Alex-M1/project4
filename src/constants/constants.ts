export const MESSAGES = {
  invalid_login: 'invalid_login',
  invalid_password: 'invalid_password',
  invalid_confirm: 'invalid_confirm',
};

export const REG_EXP = {
  login: /^[A-Za-z0-9]{1,25}/,
  password: /^[A-Za-z0-9]{6,25}$/,
};

export const COOKIE_SETTING = {
  AGE: 20000,
  SPLIT_LEN: 2,
};

export const TIC_TAC = {
  WINNER_LINES: [
    [0, 1, 2],
    [3, 4, 5],
    [6, 7, 8],
    [0, 3, 6],
    [1, 4, 7],
    [2, 5, 8],
    [0, 4, 8],
    [2, 4, 6],
  ],
  FIELD_SIZE: 480,
};
export const COOKIE_AGE = 20000;

export const GAME_TYPE = {
  TIC_TAC_TOE: 'Tic-tac-toe',
  CHECKERS: 'Checkers',
};

export const TIC_TAC_ITEM = {
  X: 'X',
  O: 'O',
};

export const CHESS_DESK = {
  COLS: [1, 2, 3, 4, 5, 6, 7, 8],
  ROWS: [1, 2, 3, 4, 5, 6, 7, 8],
  CELLS_NUM_INIT: 0,
};

export const LANGUAGE = {
  LANG: 'lang',
  EN: 'en',
  RU: 'ru',
};

export const VIEW_OPTIONS = {
  HEADER_HEIGHT: 90,
  DRAGGIN_OPACITY: {
    IS_DRAGGING: 0.3,
    NOT_DRAGGING: 1,
  },
  CHECKER_FIELD: 500,
  CELL_BACKGROUND: {
    GRAY: '#ccc',
    WHITE: '#fff',
    POSSIBLE: '#b2ff65',
  },
};

export const LOCAL_STORAGE = {
  login: 'login',
  token: 'token',
  gameOptions: 'gameOptions',
};

export const GAME_SETTINGS = {
  bot: 'Bot',
  user: 'User',
};

export const CHECKER_FIELD_INIT = [
  null,
  {
    checker: { blackChecker: false, queen: false },
    index: 1,
    blackSquare: true,
  },
  {
    checker: null,
    index: 2,
    blackSquare: false,
  },
  {
    checker: { blackChecker: false, queen: false },
    index: 3,
    blackSquare: true,
  },
  {
    checker: null, index: 4, blackSquare: false,
  },
  {
    checker: { blackChecker: false, queen: false },
    index: 5,
    blackSquare: true,
  },
  {
    checker: null,
    index: 6,
    blackSquare: false,
  },
  {
    checker: { blackChecker: false, queen: false },
    index: 7,
    blackSquare: true,
  },
  { checker: null, index: 8, blackSquare: false },
  { checker: null, index: 9, blackSquare: false },
  {
    checker: { blackChecker: false, queen: false },
    index: 10,
    blackSquare: true,
  },
  {
    checker: null,
    index: 11,
    blackSquare: false,
  },
  {
    checker: { blackChecker: false, queen: false },
    index: 12,
    blackSquare: true,
  },
  {
    checker: null,
    index: 13,
    blackSquare: false,
  },
  {
    checker: { blackChecker: false, queen: false },
    index: 14,
    blackSquare: true,
  },
  {
    checker: null,
    index: 15,
    blackSquare: false,
  },
  {
    checker: { blackChecker: false, queen: false },
    index: 16,
    blackSquare: true,
  },
  {
    checker: { blackChecker: false, queen: false },
    index: 17,
    blackSquare: true,
  },
  {
    checker: null,
    index: 18,
    blackSquare: false,
  },
  {
    checker: { blackChecker: false, queen: false },
    index: 19,
    blackSquare: true,
  }, {
    checker: null,
    index: 20,
    blackSquare: false,
  },
  {
    checker: { blackChecker: false, queen: false },
    index: 21,
    blackSquare: true,
  },
  {
    checker: null,
    index: 22,
    blackSquare: false,
  },
  {
    checker: {
      blackChecker: false,
      queen: false,
    },
    index: 23,
    blackSquare: true,
  },
  {
    checker: null,
    index: 24,
    blackSquare: false,
  },
  {
    checker: null,
    index: 25,
    blackSquare: false,
  },
  {
    checker: null,
    index: 26,
    blackSquare: true,
  },
  {
    checker: null,
    index: 27,
    blackSquare: false,
  },
  {
    checker: null,
    index: 28,
    blackSquare: true,
  },
  {
    checker: null,
    index: 29,
    blackSquare: false,
  },
  {
    checker: null,
    index: 30,
    blackSquare: true,
  },
  {
    checker: null,
    index: 31,
    blackSquare: false,
  },
  {
    checker: null,
    index: 32,
    blackSquare: true,
  },
  {
    checker: null,
    index: 33,
    blackSquare: true,
  },
  {
    checker: null,
    index: 34,
    blackSquare: false,
  },
  {
    checker: null,
    index: 35,
    blackSquare: true,
  },
  {
    checker: null,
    index: 36,
    blackSquare: false,
  },
  {
    checker: null,
    index: 37,
    blackSquare: true,
  },
  {
    checker: null,
    index: 38,
    blackSquare: false,
  },
  {
    checker: null,
    index: 39,
    blackSquare: true,
  },
  {
    checker: null,
    index: 40,
    blackSquare: false,
  },
  {
    checker: null,
    index: 41,
    blackSquare: false,
  },
  {
    checker: {
      blackChecker: true,
      queen: false,
    },
    index: 42,
    blackSquare: true,
  },
  {
    checker: null,
    index: 43,
    blackSquare: false,
  },
  {
    checker: {
      blackChecker: true,
      queen: false,
    },
    index: 44,
    blackSquare: true,
  },
  {
    checker: null,
    index: 45,
    blackSquare: false,
  },
  {
    checker: {
      blackChecker: true,
      queen: false,
    },
    index: 46,
    blackSquare: true,
  },
  {
    checker: null,
    index: 47,
    blackSquare: false,
  },
  {
    checker: {
      blackChecker: true,
      queen: false,
    },
    index: 48,
    blackSquare: true,
  },
  {
    checker: {
      blackChecker: true,
      queen: false,
    },
    index: 49,
    blackSquare: true,
  },
  {
    checker: null,
    index: 50,
    blackSquare: false,
  },
  {
    checker: {
      blackChecker: true,
      queen: false,
    },
    index: 51,
    blackSquare: true,
  },
  {
    checker: null,
    index: 52,
    blackSquare: false,
  },
  {
    checker: {
      blackChecker: true,
      queen: false,
    },
    index: 53,
    blackSquare: true,
  },
  {
    checker: null,
    index: 54,
    blackSquare: false,
  },
  {
    checker: {
      blackChecker: true,
      queen: false,
    },
    index: 55,
    blackSquare: true,
  },
  {
    checker: null,
    index: 56,
    blackSquare: false,
  },
  {
    checker: null,
    index: 57,
    blackSquare: false,
  },
  {
    checker: {
      blackChecker: true,
      queen: false,
    },
    index: 58,
    blackSquare: true,
  },
  {
    checker: null,
    index: 59,
    blackSquare: false,
  },
  {
    checker: {
      blackChecker: true,
      queen: false,
    },
    index: 60,
    blackSquare: true,
  },
  {
    checker: null,
    index: 61,
    blackSquare: false,
  },
  {
    checker: {
      blackChecker: true,
      queen: false,
    },
    index: 62,
    blackSquare: true,
  },
  {
    checker: null,
    index: 63,
    blackSquare: false,
  },
  {
    checker: {
      blackChecker: true,
      queen: false,
    },
    index: 64,
    blackSquare: true,
  },
];
