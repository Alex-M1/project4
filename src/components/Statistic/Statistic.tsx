import React from 'react';
import { IStatUUID } from 'store/statistic/types';
import { useTheme } from '../hooks/useTheme';
import { StContainer } from './styled';

interface IProps {
  getStatUUID: IStatUUID[],
}

const Statistic: React.FC <IProps> = () => {
  const { colors, theme } = useTheme();
  return (
      <StContainer
        theme={theme}
        colors={colors}
      >
        Statistic will be here
      </StContainer>
  );
};
export default Statistic;
