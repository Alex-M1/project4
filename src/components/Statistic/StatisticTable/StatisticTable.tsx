import React, { useEffect } from 'react';
import { useTranslation } from 'react-i18next';
import { v4 as uuidv4 } from 'uuid';
import { useTheme } from 'src/components/hooks/useTheme';
import { IStatUUID } from 'store/statistic/types';
import StatisticItem from '../StatisticItem';
import { StHeaderWrapper, StResultsWrapper, StWrapper } from './styled';

interface IProps {
  getStatistic: () => void;
  statisticData: Array<IStatUUID>;
}

const StatisticTable: React.FC <IProps> = ({ getStatistic, statisticData }) => {
  useEffect(() => {
    getStatistic();
  }, []);
  const { t } = useTranslation();
  const { colors, theme } = useTheme();
  const statistic = statisticData.map((el) => {
    return <StatisticItem {...el} key={uuidv4()}/>;
  });

  return (
    <StWrapper
          theme={theme}
          colors={colors}
    >
      <StHeaderWrapper
        theme={theme}
        colors={colors}
      >
        <h1>{t('game_statistic')}</h1>
        <div>
          <span>{t('user_name')}</span>
          <span>{t('game_type')}</span>
          <span>{t('opposer')}</span>
          <span>{t('winner')}</span>
        </div>
      </StHeaderWrapper>
      <StResultsWrapper
        theme={theme}
        colors={colors}
      >
        {statistic}
      </StResultsWrapper>
    </StWrapper>
  );
};

export default StatisticTable;
