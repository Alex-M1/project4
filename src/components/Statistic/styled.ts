import styled from 'styled-components';
import { IStyled } from 'common/types/styledTypes';

export const StContainer = styled.div <IStyled>`
  display: flex;
  justify-content: space-around;
  align-items: center;
  width: 100%;
  height: 85%;
`;
