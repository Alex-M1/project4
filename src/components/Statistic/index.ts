import { connect } from 'react-redux';
import { AppStateType } from 'store/rootReducer';
import { getStatisticByUuid } from 'store/statistic/selectors';
import Statistic from './Statistic';

const mapStateToProps = (state: AppStateType) => ({
  getStatUUID: getStatisticByUuid(state),
});

export default connect(mapStateToProps)(Statistic);
