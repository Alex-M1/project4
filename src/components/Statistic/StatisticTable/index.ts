import { connect } from 'react-redux';
import { AppStateType } from 'store/rootReducer';
import { getStatistic } from 'store/statistic/action';
import { getStatisticByUuid } from 'store/statistic/selectors';
import StatisticTable from './StatisticTable';

const mapStateToProps = (state: AppStateType) => ({
  statisticData: getStatisticByUuid(state),
});
const mapDispatchToProps = {
  getStatistic,
};

export default connect(mapStateToProps, mapDispatchToProps)(StatisticTable);
