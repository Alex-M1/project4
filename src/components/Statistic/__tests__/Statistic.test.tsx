import { GAME_TYPE } from 'constants/constants';
import React from 'react';
import configureStore from 'redux-mock-store';
import { shallowSmart, mountSmart } from 'src/__tests__/testHelper';
import Statistic from '../Statistic';
import StatisticTable from '../StatisticTable';

jest.mock('react-i18next', () => ({
    useTranslation: jest.fn().mockImplementation(() => ({
      t: jest.fn(),
    })),
  }),
);

const mockStore = configureStore();
const store = mockStore({
  room: {
    rooms: [],
    gameType: GAME_TYPE.CHECKERS,
    toRoom: '',
  },
  ticTac: {
    squares: Array(9).fill(null),
    steps: {
      count: 0,
      isUserStep: null,
    },
    isGameEnd: false,
    isMyTurn: true,
    myOpponentGame: {
      id: null,
      gameType: null,
      guestLogin: null,
      startTime: null,
      stepDtoList: null,
    },
    winner: '',
  },
  statistic: {
    statistic: [
      {
        creatorLogin: 'string',
        gameType: 'string',
        guestLogin: 'string',
        winnerLogin: 'string',
      },
    ],
  },
});
describe('Statistic', () => {
  it('Should match snapshot', () => {
    const component = shallowSmart(<Statistic />, store);
    expect(component.html()).toMatchSnapshot();
  });
  it('should render StContainer', () => {
    const component = mountSmart(<Statistic />, store);
    expect(component.find('styled__StContainer')).toHaveLength(1);
  });
  it('should render StatisticTable', () => {
    const component = mountSmart(<StatisticTable />, store);
    expect(component.find('StatisticTable')).toHaveLength(1);
  });
});
