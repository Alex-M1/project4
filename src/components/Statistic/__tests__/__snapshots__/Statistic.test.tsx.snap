// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Statistic Should match snapshot 1`] = `"<div class=\\"styled__StContainer-rIpsz kMbkNV\\"><div class=\\"styled__StWrapper-kHlHLr fYkXuW\\"><div class=\\"styled__StHeaderWrapper-bKzUbg fqprgP\\"><h1></h1><div><span></span><span></span><span></span><span></span></div></div><div class=\\"styled__StResultsWrapper-eOGtIh gxuytl\\"><div><span></span><span></span><span>string</span><span>string</span></div></div></div></div>"`;
